### ISSUE TITLE
Pool plots without replotting

### ISSUE THREAD

**Issue by seertenedos** [madMAx43v3r/mmx-node#40]
I know you stated pooling would come later wich is great but is it possible to make it so that we could generate plots now that could be used with a pool later on when it is added vs needing to replot everything like we did with chia. Personally i found that to be a huge waste of resources that upfront planing could have prevented.

**Comment by scrutinously** [comment]
The way current contract addresses are generated is from the transaction that they are created in. If there were to be a system similar to the chia plotnft that uses the existing contract system, they wouldn't be compatible over different blockchains (going from testnet to testnet, or testnet to mainnet) because the hash for the transaction wouldn't exist there. It's certainly possible that Max could come up with some brilliant new way to create "portable" plots, but I would suggest setting your expectations to the the idea that you will have OG plots to farm with at the start, and can slowly replot to "portable" plots after mainnet launch if you want. One way to minimize your resource usage would be to plot all of your OG as k30, then slowly replot portables to k32 later.

**Comment by seertenedos** [comment]
honestly i was hoping that part of the plot with the farm key and and the contract key would be improved.  maybe it can't be done but some way of either brining the contracts over or allowing the contract key to be set later. Myabe a way to have 2 keys per plot now a farmer key and contract key that somehow can be bound to a contract using the farmer private key but only done once or something like that.

i never did really understand why the farmer and contract keys were so locked into the plot. i get others steally your plots etc would be bad but it is not like yo could really multi farm with them as the plot key i thought was used as part of reward claiming i though.  It does help prevent people selling the plots and potentially selling them twice to 2 different people so maybe that was the reason but i would have thought that was a minority case and potentially solved with a smart contract to that has the registered owner etc.

**Comment by scrutinously** [comment]
Yeah, would be interesting to have a transferrable contract that the plot points to, however that would still need to be on the blockchain so couldn't be minted until mainnet anyway. The reason the keys are baked into the plot is for the security of the proof that comes out of the plot. As I said earlier, the reason you currently can't "import" a contract onto a new mmx blockchain is due to the way that the contract address is generated. No telling if max will change that later, so anything is possible. Will just have to wait and see.

I'm moving this to discussions because it's not really an issue with mmx node, we should keep this area clear for actual problems in the code that need attention.
